02-spatial-data.qmd

```diff
@@ -185,41 +192,79 @@ point
 
 Here is an example of a `MULTIPOINT` geometry:
 
+```{python}
+from shapely.geometry import MultiPoint
+multipoint = MultiPoint([(5,2), (1,3), (3,4), (3,2)])
+multipoint
+```
+
 ```{python}
 multipoint = wkt.loads("MULTIPOINT ((5 2), (1 3), (3 4), (3 2))")
 multipoint
 ```
 
 Here is an example of a `LINESTRING` geometry:
 
+```{python}
+from shapely.geometry import LineString
+linestring = LineString([(1,5), (4,4), (4,1), (2,2), (3,2)])
```

Comment:
This is neat! Out of interest is there a way to do that from a matrix? In R it would be:

``` r
sf::st_linestring(matrix(c(1, 5, 4, 4, 4, 1, 2, 2, 3, 2), ncol = 2, 
    byrow = TRUE))
#> LINESTRING (1 5, 4 4, 4 1, 2 2, 3 2)
```

<sup>Created on 2022-04-08 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

Reply:
`LineString` can accept an array with two columns too (I guess because it's considered a list of rows):

```python
from shapely.geometry import LineString
import numpy as np
x = np.array([(1,5), (4,4), (4,1), (2,2), (3,2)])
print(LineString(x))
## LINESTRING (1 5, 4 4, 4 1, 2 2, 3 2)
```

Reply:
Fantastic, that is good to know, and seems that geometries are easy (easier?) to create in Python?

Reply:
Yes, seems to me too

Reply:
Yes, it's pretty easy. The only not so intuitive constructor is the Polygon with holes, imho. 

In these examples, I used the list of tuples representation but it also works with list of lists and - as Michael  showed - with np.array.